releasenotes/notes/0.23/remove_mcmt_label-cee8a11e0164f8e1.yaml

```
@@ -1,4 +1,20 @@
 ---
 upgrade:
   - |
-    The classes :class:`~.MCMT` and :class:`~.MCMTVChain` deprecated the argument ``label`` in Terra 0.19, and it has now been removed.
+    The previously deprecated ``label`` argument for the constructor of the
+    :class:`~.MCMT` and :class:`~.MCMTVChain` classes has been removed.
+    It was deprecated as of the Qiskit Terra 0.19.0 release (Decemeber, 2021).
+    Using the ``label`` argument on these classes was undefined behavior
+    as they are subclasses of :class:`~.QuantumCircuit` instead of
+    :class:`~.circuit.Instruction`. This would result in the assigned label generally
+    being ignored. If you need to assign a ``label`` to an
+    instance of :class:`~.MCMT` or :class:`~.MCMTVChain` you should convert
+    them to an :class:`~.Gate` instance with :meth:`~.QuantumCircuit.to_gate`
+    and then assign the desired label to :attr:`~.Gate.label` attribute. For
+    example::
+
+        from qiskit.circuit.library import MCMT, XGate
+
+        mcmt_circuit = MCMT(XGate(), 3, 2)
+        mcmt_gate = mcmt_circuit.to_gate()
+        mcmt_gate.label = "Custom MCMT X"
```

Technically you're right to include this, but in practice, `label` is something we're going to want to heavily de-emphasise in the move to singletons.  This gate probably won't end up as a singleton, but even so, we _maybe_ want to drop this bit?

```suggestion
    being ignored.
```

---

The example wasn''t in there originally and I was struggling to visualize exactly what a user doing `MCMT(..., label="")` was meant to do with just the text which is why I added the example. I don't really feel strongly one way or the other though, I think this is a definitely and edge use case. But I just generally feel upgrade notes should state the alternative if one is available (or the justification why one isn't).

---

I'm happy to leave this on the basis of providing complete release notes tbh.